<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Lua4RSMainWidget</class>
 <widget class="QWidget" name="Lua4RSMainWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QLabel" name="l_titel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>701</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>18</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>Lua4RS - by far*call &amp; sehraf</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="pte_code">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>701</width>
     <height>351</height>
    </rect>
   </property>
   <property name="horizontalScrollBarPolicy">
    <enum>Qt::ScrollBarAsNeeded</enum>
   </property>
   <property name="lineWrapMode">
    <enum>QPlainTextEdit::NoWrap</enum>
   </property>
   <property name="plainText">
    <string>function getName( id )
	return getPeerName( id )
end

function tablelength( T )
	local count = 0
	for _ in pairs(T) do count = count + 1 end
	return count
end

function showInfo()
	rs_clear()
	rs_print( &quot;my SSL ID:&quot; )
	rs_print( getOwnId() )
	rs_print( &quot;&quot; )

	all, online = getPeerCount() 
	rs_print( &quot;I have &quot; .. all .. &quot; friends amd &quot; .. online .. &quot; are online&quot; )
	rs_print( &quot;&quot; )

	friends = getFriendList()
	rs_print( &quot;my friends are:&quot; )
	if #friends == 0 then
		rs_print( &quot;i don't have any :(&quot; )
	elseif #friends &lt;= 20 then
		for i = 1, #friends do
			rs_print( getName ( friends[i] ) )
		end
	else
		rs_print( &quot;way too many ...&quot; )
	end

	rs_print( &quot;&quot; )

	onlinePeers = getOnlineList()
	rs_print( &quot;online are:&quot; )
	if #onlinePeers == 0 then
		rs_print( &quot;none :(&quot; )
	elseif tablelength( onlinePeers ) &lt;= 20 then
		for i = 1, #onlinePeers do
			rs_print( getName( onlinePeers[i] ) )
		end
	else
		rs_print( &quot;too many :S&quot; )
	end
end

function showDetails()
	rs_clear()
	
	details = getPeerDetails( getOwnId() )
	for key,value in pairs(details) do
		rs_print( key .. &quot;: &quot; .. value )
	end
end</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="pte_output">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>410</y>
     <width>700</width>
     <height>181</height>
    </rect>
   </property>
   <property name="horizontalScrollBarPolicy">
    <enum>Qt::ScrollBarAsNeeded</enum>
   </property>
   <property name="lineWrapMode">
    <enum>QPlainTextEdit::NoWrap</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="pb_run">
   <property name="geometry">
    <rect>
     <x>900</x>
     <y>10</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>run Lua</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="pte_hints">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>50</y>
     <width>271</width>
     <height>541</height>
    </rect>
   </property>
   <property name="horizontalScrollBarPolicy">
    <enum>Qt::ScrollBarAsNeeded</enum>
   </property>
   <property name="lineWrapMode">
    <enum>QPlainTextEdit::NoWrap</enum>
   </property>
   <property name="plainText">
    <string>rs_print( string)
rs_clear()
string getOwnId()
array getOnlineList()
array getFriendList()
int(all), int(online) getPeerCount()
string getPeerName( string )
table getPeerDetails( string )</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
